jobboard/tests.py

```diff
@@ -182,3 +182,41 @@ def test_page_display_to_user_that_is_recruiter(client):
     jobs_to_display = Job.get_jobs_posted_on_or_after_specific_date(date_six_months_ago)
     # test that jobs displayed are only jobs created 6 months ago
     assert response.context['jobs_to_display'] == jobs_to_display
+
+
+@pytest.mark.django_db
+def test_navbar_links_are_correct_for_student_user(client, example_student_a):
+    client.force_login(example_student_a.user)
+    response = client.get("/")
+    assert response.status_code == 200
+    assert response.context['navbar_links'] == {"Account Settings": "/student/account_settings/", "Logout": "/logout",
+                                                f"Welcome {example_student_a.user.username}": "#"}
+
+
+@pytest.mark.django_db
+def test_navbar_links_are_correct_for_recruiter_user(client):
+    recruiter_user_from_db = Recruiter.objects.get(name='a').user
+    client.force_login(recruiter_user_from_db)
+    response = client.get("/")
+    assert response.status_code == 200
+    assert response.context['navbar_links'] == {"Account Settings": "/recruiter/account_settings/",
+                                                "My Jobs": "/myjobs",
+                                                "Create Job": "/recruiter/create_new_job_form", "Logout": "/logout",
+                                                f"Welcome {recruiter_user_from_db.username}": "#"}
+
+
+@pytest.mark.django_db
+def test_navbar_links_are_correct_for_not_logged_in_user(client):
+    response = client.get("/")
+    assert response.status_code == 200
+    assert response.context['navbar_links'] == {"Login": "/login"}
+
+
+@pytest.mark.django_db
+def test_navbar_links_are_correct_for_logged_in_user_not_recruiter_or_student(client):
+    test_user = User.objects.create_user(username="testuser1", password="testuser1")
+    test_user.save()
```

Comment:
why are you saving the test_user in the DB?

Reply:
@guy9050 
Each test creates a temporary DB file on which it is running, thus saving an instance in the DB inside a test doesn't actually save it in the DB (only in the temp DB it's using for the test).